ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/Amount.kt

```diff
@@ -0,0 +1,366 @@
+     * @return Returns an [Amount] representing the difference between the specified quantity subtracted from the current quantity.
+     */
+    operator fun minus(value: Int): Amount<T> {
+        return minus(value.toBigInteger())
+    }
+
+    /**
+     * Subtracts the specified amount from the current [Amount].
+     *
+     * @param value The amount to subtract from the current [Amount].
+     * @return Returns an [Amount] representing the difference between the specified quantity subtracted from the current quantity.
+     */
+    operator fun minus(value: Long): Amount<T> {
+        return minus(value.toBigInteger())
+    }
+
+    /**
+     * Subtracts the specified amount from the current [Amount].
+     *
+     * @param value The amount to subtract from the current [Amount].
+     * @return Returns an [Amount] representing the difference between the specified quantity subtracted from the current quantity.
+     */
+    operator fun minus(value: Amount<T>): Amount<T> {
+        checkItem(value)
+        return minus(value.quantity)
+    }
+
+    /**
+     * Multiplies the specified amount by the current [Amount].
+     *
+     * @param value The amount to multiply by the current [Amount].
+     * @return Returns an [Amount] representing the product of the specified quantity multiplied by the current quantity.
+     */
+    operator fun times(value: BigInteger): Amount<T> {
+        return Amount(quantity * value, item)
+    }
+
+    /**
+     * Multiplies the specified amount by the current [Amount].
+     *
+     * @param value The amount to multiply by the current [Amount].
+     * @return Returns an [Amount] representing the product of the specified quantity multiplied by the current quantity.
+     */
+    operator fun times(value: Int): Amount<T> {
+        return times(value.toBigInteger())
+    }
+
+    /**
+     * Multiplies the specified amount by the current [Amount].
+     *
+     * @param value The amount to multiply by the current [Amount].
+     * @return Returns an [Amount] representing the product of the specified quantity multiplied by the current quantity.
+     */
+    operator fun times(value: Long): Amount<T> {
+        return times(value.toBigInteger())
+    }
+
+    /**
+     * Multiplies the specified amount by the current [Amount].
+     *
+     * @param value The amount to multiply by the current [Amount].
+     * @return Returns an [Amount] representing the product of the specified quantity multiplied by the current quantity.
+     */
+    operator fun times(value: Amount<T>): Amount<T> {
+        checkItem(value)
+        return times(value.quantity)
+    }
+
+    /**
+     * Divides the current [Amount] by the specified amount.
+     *
+     * @param value The amount to divide the current [Amount] by.
+     * @return Returns an [Amount] representing the quotient of the current quantity divided by the specified quantity.
+     */
+    operator fun div(value: BigInteger): Amount<T> {
```

Comment:
I intentionally did not provide div in the original, because it won't conserve quantity, hence the splitEvenly.

Reply:
Happy to remove `div` and `rem` if they are useless. Or we could keep them and make them private for now, and use them under the hood to implement `splitEvenly`? 

If then in the (rather unlikely) case that a customer asked why `Amount` is not divisible, we could either offer `splitEvenly`, or make `div` and `rem` public in Corda 5.x should there be any push for it.